spring-integration-file/src/main/java/org/springframework/integration/file/remote/synchronizer/AbstractInboundFileSynchronizer.java

```diff
@@ -342,18 +343,42 @@ protected void copyFileToLocalDirectory(String remoteDirectoryPath, F remoteFile
 				}
 			}
 
-			if (tempFile.renameTo(localFile)) {
+			boolean renamed = tempFile.renameTo(localFile);
+
+			if (!renamed) {
+				if (localFile.delete()) {
+					renamed = tempFile.renameTo(localFile);
+				}
+			}
```

Comment:
We should log an ERROR is `localFile.delete` fails or the second `renameTo` fails the logging below does not provide enough information as to the root cause.